src/main/webapp/index.html

```diff
@@ -50,7 +50,7 @@
       <h1 class="logo">smallcity</h1>
       <button class="geolocate-button" onclick="getGeolocation()"><i class="fas fa-location-arrow"></i> locate me</button>
       <div class="zipcode-container">
-        your zipcode:
+        city, state, or zip:
```

Comment:
Is this requesting all of these bits, or some sub combination? What happens if a user only enters the city? What happens if the user only enters the state? What happens if they enter both? etc.

Reply:
It should work if the user enters any combination of those things. I'm not sure how to express that on the front end though...something like "enter your location" or "where are you?" Do you have any suggestions?

Reply:
You can add filler text inside the zip code search box, to have "Zip code | City | State" and that should be clear for the user.

Reply:
What about cities with the same name, but in different states? What will this return? It seems like one will just be chosen by the api (best case scenario). I think what you have now works fine, but these error cases should be appropriately handled in the code.

Reply:
If there is a cities with the same name, there wouldn't be an issue if the user also entered an addition identifier (like state). If not, I believe that the API would just choose one (ex. if I search "Springfield", the results will be Springfield, MO, but if I search "Springfield, IL" the results will be in Illinois). I'm having a bit of trouble thinking about how these cases can be handled if we don't know what the user wants their results to be. Do you have any suggestions?

Reply:
Before we go there, what benefit is there to allowing users to enter a city, state combination. Results can be returned for presumably anywhere (defined by the API) in that city. These can be outside of the defined radius for the application. Is this outside of the core scope for you app?

Reply:
I don't think that it is. If a user is travelling, they likely won't know the zipcode of the place that they're travelling to off of the top of their head. I think the benefit of the city, state combination is allowing the user to get less specific results while being able to reference a specific city. I think this feature would be more beneficial if we were displaying more than 15 results, but I still think that the use case is valuable.

Reply:
I agree. This is a useful thing to support. My concern however, is that the behaviour seems undefined. i.e. it's not clear what the focal point becomes with only a city & state input pair. Perhaps this is documented in the api docs somewhere?

Reply:
The results of the Geocoding API returns a parameter called ["locationType"](https://developers.google.com/maps/documentation/geocoding/overview#Results) which seems to specify the accuracy of the result. I tested the code a few times, and it seems like "Approximate" is returned for both zipCodes and city/state combinations. When an exact address is entered, it returns "Rooftop" which indicates a precise geocode. One of the other return options is "Geometric center" which to me appears to be what is chosen when a broad input is passed because it looks like it's centering around the location.

Reply:
Ok. You should be able to confirm this in the results payload JSON array. Let's document these various return types in your design doc.

Reply:
I added a logger message to track the location type and I will add to the design doc.